sympy/ntheory/primetest.py

```diff
@@ -643,3 +643,25 @@ def isprime(n):
     # Add a random M-R base
     #import random
     #return mr(n, [2, random.randint(3, n-1)]) and is_strong_lucas_prp(n)
+
+
+def is_gaussian_prime(num):
+    """Test if num is a Gaussian prime number.
+
+    References
+    ==========
+
+    .. [1] https://oeis.org/wiki/Gaussian_primes
+    """
+    from sympy import I
+    num += 0*I
```

Comment:
Why is this needed?

Reply:
When num is integr, it shows `AttributeError: 'int' object has no attribute 'as_real_imag'`

Reply:
I think that it's better to use sympify. There is no advantage of doing some sympify hacks like this, because addition already does sympify internally.